twib/twib/Twib.cpp

```diff
@@ -60,23 +110,32 @@ Twib::~Twib() {
 }
 
 void Twib::event_thread_func() {
+	fd_set recvset;
+	fd_set sendset;
+	int maxfd = 0;
 	while(!event_thread_destroy) {
 		log(DEBUG, "event thread loop");
 
-		struct pollfd pollfds[2];
-		pollfds[0] = {.fd = event_thread_notification_pipe[0], .events = POLLIN};
-		pollfds[1] = {.fd = fd, .events = POLLIN};
+		FD_ZERO(&recvset);
+		FD_ZERO(&sendset);
+#ifndef _WIN32
+		FD_SET(event_thread_notification_pipe[0], &recvset);
+		maxfd = std::max(maxfd, event_thread_notification_pipe[0]);
+#endif
+		FD_SET(fd, &recvset);
+		maxfd = std::max(maxfd, fd);
 		if(out_buffer.ReadAvailable() > 0) {
-			pollfds[1].events|= POLLOUT;
+			FD_SET(fd, &sendset);
 		}
 
-		if(poll(pollfds, sizeof(pollfds)/sizeof(pollfds[0]), -1) < 0) {
-			log(FATAL, "failed to poll file descriptors: %s", strerror(errno));
+		if(select(maxfd + 1, &recvset, &sendset, NULL, NULL) < 0) {
```

Comment:
I'm not really a big fan of having to use select. I think that we should have a `SocketFrontend` class that `UNIXFrontend` and `TCPFrontend` inherit from, and have the Linux/OSX implementation of `SocketFrontend` use `poll` and the Windows implementation use `WSAWaitForMultipleEvents`. Some people on [stackoverflow](https://stackoverflow.com/questions/3333361/how-to-cancel-waiting-in-select-on-windows) say you can use `QueueUserAPC` to interrupt this, which provides a solution for the event notification pipe problem.

Just notes for another day though. Not a big enough deal to stop this PR's merge.

Reply:
Or perhaps we could use WSAPoll?